### PR TITLE
Implement new approaches based on JS SDK and React SDK

### PR DESCRIPTION
**Pull request by schabibi1**
New implementations from Storyblok side
- [x] storyblokInit and apiPlugin for initializing Storyblok API
- [x] StoryblokCmponent for conditionally handling dynamic components
- [x] storyblokEditable to make components being editable
- [x] useStoryblokState to handle adding events on the visual editor and loading Storyblok JS Bridge & parsing JSON

New configuration for tests and others
- [x] implemented Cypress
- [x] set up 2 playgrounds - playground & playground-test
- [x] set up dependabot
- [x] update docs

### CODE REVIEW

**Review by alexjoverm — CHANGES_REQUESTED** [review]
Please add the commit to release the major version (the one starting with feat: and the BREAKING CHANGE: in the next line)

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { useEffect, useState } from "react";
+import { useStoryblokBridge as useSbBridge } from "@storyblok/react";
```
Comment (fgiuliani) [inline lib/index.js:2]: Do we need to import useStoryblokBridge with an alias?
useStoryblokBridge is already being imported/exported in the line below. Can we use that same instance of the function?

File: lib/package.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{
+  "name": "gatsby-source-storyblok",
+  "contributors": [
+    {
+      "name": "Alexander Feiglstorfer <delooks@gmail.com>"
+    }
+  ],
```
Comment (fgiuliani) [inline lib/package.json:7]: We can remove this block

File: lib/package.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    "test:e2e": "start-server-and-test cy:playground http-get://localhost:8000/ cy:run",
+    "test:e2e-watch": "start-server-and-test cy:playground http-get://localhost:8000/ cy:open",
+    "cy:playground": "npm run develop --prefix ../playground-test",
+    "cy:run": "cypress run",
+    "cy:open": "cypress open",
+    "prepublishOnly": "npm run build && cp ../README.md ./"
+  },
+  "dependencies": {
+    "@storyblok/react": "^1.0.5",
+    "storyblok-js-client": "^4.5.2"
```
Comment (fgiuliani) [inline lib/package.json:31]: I think we don't need this dependency. The JS client is already included in `@storyblok/js` and, because of that, it's also included in `@storyblok/react`

File: lib/cypress/component/index.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+describe("gatsby-source-storyblok", () => {
+  beforeEach(() => {
+    cy.spy(window.console, "log").as("log");
+    cy.spy(window.console, "error").as("error");
+    delete window.storyblokRegisterEvent;
+    document.getElementById("storyblok-javascript-bridge")?.remove();
+  });
+
+  describe("JS Bridge", () => {
```
Comment (fgiuliani) [inline lib/cypress/component/index.spec.jsx:18]: Question about the tests:

I see that the only function created in `lib/index.js` is `useStoryblokState`, but in the file `lib/cypress/component/index.spec.jsx` I see tests related to "useStoryblok", the StoryblokComponent, and the JS Bridge.

Are these tests different from the ones we already have in `@storyblok/react`?
https://github.com/storyblok/storyblok-react/blob/feature/v1/lib/cypress/component/index.spec.jsx

File: lib/testing-components/Test.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    bridge,
+    use: accessToken ? [apiPlugin] : [],
+    components,
+  });
+
+  const storyblokApi = getStoryblokApi();
+  const apiExists = !!(storyblokApi && typeof storyblokApi.get === "function");
+
+  useEffect(() => {
+    useStoryblokBridge(43423, (newStory) => console.log(newStory));
```
Comment (fgiuliani) [inline lib/testing-components/Test.jsx:22]: Should we test `useStoryblokState` here?
Is this `useStoryblokBridge` function different from the one from `@storyblok/react`?

File: lib/testing-components/TestUseStoryblok.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import {
```
Comment (fgiuliani) [inline lib/testing-components/TestUseStoryblok.jsx:1]: Do we need this test in this SDK?

File: playground-test/components/feature.jsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
```
Comment (fgiuliani) [inline playground-test/components/feature.jsx:1]: I'm probably missing something.
What's the difference between `playground-test` and `playground`?

Also, I see the components appearing twice in `playground-test`.
We have
`playground-test/components/feature.jsx`
and also
`playground-test/src/components/Feature.js`
Do we need both?

File: lib/index.js (hunk 1 above)
Comment (schabibi1) [reply lib/index.js:2]: I believe yes as it shows as not defined if I don't use the alias as in this screenshot.

```javascript
// what I tested

import { useEffect, useState } from "react";
// 👇 excluded the alias
// import { useStoryblokBridge as useSbBridge } from "@storyblok/react";
export { useStoryblokBridge, storyblokInit, apiPlugin, StoryblokComponent, storyblokEditable } from "@storyblok/react";

export function useStoryblokState(originalStory, bridgeOptions) {
  if (typeof originalStory.content === "string") originalStory.content = JSON.parse(originalStory.content);

  let [story, setStory] = useState(originalStory);
  useEffect(() => {
    // 👇 changed from alias to useStoryblokBridge
    useStoryblokBridge(story.internalId, (newStory) => setStory(newStory), bridgeOptions);
  }, []);

  return story;
}
```

Screenshot 2022-06-05 at 15 54 47

File: lib/package.json (hunk 2 above)
Comment (schabibi1) [reply lib/package.json:7]: I think @alexjoverm and I added it to see if it passes when we trigger the GitHub actions. I can delete if the other SDKs don't contain👍

File: lib/package.json (hunk 3 above)
Comment (schabibi1) [reply lib/package.json:31]: I believe we still need @storyblok/react in dependency as Gatsby SDK needs to import the JS Client and other APIs coming from JS and React SDKs. What do you think?

File: lib/cypress/component/index.spec.jsx (hunk 4 above)
Comment (schabibi1) [reply lib/cypress/component/index.spec.jsx:18]: Good catch! Also, in Gatsby SDK, it's `useSToryblokState` instead of `useStoryblok`.

 @alexjoverm and I had a conversation that we don't need to test common APIs that are coming from React SDK, which means that we don't need to duplicate the test.

I'll fix it to make sense.

File: lib/testing-components/Test.jsx (hunk 5 above)
Comment (schabibi1) [reply lib/testing-components/Test.jsx:22]: I think it could be considered the same except that you just need to be careful about what to store as an id in the first param.

(Gatsby's `id` is String but `internalId` from Gatsby retrieves the id format we need (Number).)
If it's the same as React SDK, we don't need it, don't we? (But Gatsby SDK uses an alias. Do we need it in this case?)🤔

File: lib/testing-components/TestUseStoryblok.jsx (hunk 6 above)
Comment (schabibi1) [reply lib/testing-components/TestUseStoryblok.jsx:1]: Hmm I bet we don't need it. Good catch! @alexjoverm what do you think?

File: lib/index.js (hunk 1 above)
Comment (alexjoverm) [reply lib/index.js:2]: Yes. The line below is `export from` so you still need to import it if you want to use it

File: lib/package.json (hunk 2 above)
Comment (alexjoverm) [reply lib/package.json:7]: Yep, Facundo is right

File: lib/package.json (hunk 3 above)
Comment (alexjoverm) [reply lib/package.json:31]: I think Facundo means storyblok-js-client. Normally that'd be true, but in this case it's used by `gatsby-node.js` which is a node module and we'd need a bit of refactoring to remove the dep

File: lib/cypress/component/index.spec.jsx (hunk 4 above)
Comment (alexjoverm) [reply lib/cypress/component/index.spec.jsx:18]: 👍

File: lib/testing-components/Test.jsx (hunk 5 above)
Comment (alexjoverm) [reply lib/testing-components/Test.jsx:22]: Yeah... Gatsby do some weird things so Gatsby SDK's `useStoryblokState` differs a bit from the react one

File: lib/testing-components/TestUseStoryblok.jsx (hunk 6 above)
Comment (alexjoverm) [reply lib/testing-components/TestUseStoryblok.jsx:1]: True!

File: playground-test/components/feature.jsx (hunk 7 above)
Comment (alexjoverm) [reply playground-test/components/feature.jsx:1]: That was a conscious decision. Playground test is not meant to change (unless you change the tests) and meant for testing. While playground is meant to mess around

File: lib/index.js (hunk 1 above)
Comment (fgiuliani) [reply lib/index.js:2]: Oooh that's interesting. Thanks for the insights @schabibi1 @alexjoverm

File: lib/package.json (hunk 3 above)
Comment (fgiuliani) [reply lib/package.json:31]: Yes, as @alexjoverm says I was talking about `storyblok-js-client`.

Thanks for the clarification, I didn't see it being used in `gatsby-node.js` before

File: lib/testing-components/Test.jsx (hunk 5 above)
Comment (fgiuliani) [reply lib/testing-components/Test.jsx:22]: 🚀

File: playground-test/components/feature.jsx (hunk 7 above)
Comment (fgiuliani) [reply playground-test/components/feature.jsx:1]: Hey @alexjoverm 
What does it mean to "mess around" with `playground`?
On the other hand, how is `playground-test` being used in the tests?

Besides that, is there any reason why we need duplicate components in `playground-test`? We have a `playground-test/components/`and a `playground-test/src/components/` folders, and the components inside of them look kind of the same.

Comment (alexjoverm) [reply playground-test/components/feature.jsx:1]: True, not necessary to duplicate those components (might be leftovers of copy/paste).

About why need both... Maybe in next meeting?

Comment (fgiuliani) [reply playground-test/components/feature.jsx:1]: Nice! 😄

Comment (schabibi1) [reply playground-test/components/feature.jsx:1]: Het @alexjoverm and @fgiuliani 
Good catch again! I'll remove `playground-test/components` and `playground/components` as these files are duplications by mistake👍

Include me in the next meeting about that, Alex🙏

**Review by schabibi1 — COMMENTED** [review]
Done👍

**Review by alexjoverm — APPROVED** [review]

**Review by fgiuliani — APPROVED** [review]
